addons/hr_holidays/models/hr_leave.py

```diff
@@ -394,6 +422,9 @@ def action_validate(self):
         if not self.env.user.has_group('hr_holidays.group_hr_holidays_user'):
             raise UserError(_('Only an HR Officer or Manager can approve leave requests.'))
 
+        if any(not holiday.can_approve for holiday in self):
```

Comment:
Error message does not match can_approve implementation as it is not only linked to my own request.